pkg/connect/gateway.go

```diff
@@ -0,0 +1,435 @@
+package connect
+
+import (
```

Comment:
this file is just splitting out the gateway related logic out of the generic service functions.

Reply:
any way to split this to preserve git history? np if not